Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -1029,7 +1056,7 @@ namespace xr
         }
 
     private:
-        bool isInitialized{false};
+        bool& isInitialized;
```

Comment:
>bool& isInitialized; [](http://example.com/codeflow?start=8&length=20)

Rather than store these directly as references, would it just make more sense to remove the member itself and always call into the context object instead? Member references are often difficult to reason about object lifetimes for.

Reply:
seems reasonable, let me update this code :)

Reply:
fwiw, the system impl is already obtained through a reference like this so going through it seems to have this same potential risk. I'm going to change the context to a shared pointer so the session and system can have their own individual references

Reply:
the change is much larger now in scope, but we now have the value backed by a shared pointer that will exist for the whole session lifecycle even if the system gets torn down for some reason